packages/documentation-v7/src/stories/components/badge/badge.stories.tsx

```diff
@@ -0,0 +1,229 @@
+import { html, nothing } from 'lit';
+import { BADGE } from '../../../../.storybook/constants';
+import { mapClasses } from '../../../utils';
+
+const meta: Meta = {
+  title: 'Components/Badge',
+  render: renderBadge,
+  decorators: [externalControl],
+  parameters: {
+    badges: [BADGE.BETA, BADGE.NEEDS_REVISION],
+  },
+  args: {
+    text: 'Insigno',
+    size: 'default',
+    nestedBadge: false,
+    interactionType: 'none',
+    checked: false,
+    dismissed: false,
+  },
+  argTypes: {
+    text: {
+      name: 'Text',
+      description: 'The text contained in the badge.',
+      control: {
+        type: 'text',
+      },
+      table: {
+        category: 'Content',
+      },
+    },
+    size: {
+      name: 'Size',
+      description: 'The size of the badge.',
+      control: {
+        type: 'radio',
+        labels: {
+          'default': 'Default',
+          'badge-sm': 'Small',
+        },
+      },
+      options: ['default', 'badge-sm'],
+      table: {
+        category: 'General',
+      },
+    },
+    nestedBadge: {
+      name: 'Nested Badge',
+      description: 'If `true`, a nested badge is displayed inside the main badge.',
+      control: {
+        type: 'boolean',
+      },
+      table: {
+        category: 'General',
+      },
+    },
+    interactionType: {
+      name: 'Interaction Type',
+      description: 'Defines how the badge can be interacted with.',
+      control: {
+        type: 'inline-radio',
+        labels: {
+          'none': 'None',
+          'checkable': 'Checkable',
+          'dismissible': 'Dismissible',
+        },
+      },
+      options: ['none', 'checkable', 'dismissible'],
+      table: {
+        category: 'Interactions',
+      },
+    },
+    checked: {
+      name: 'Checked',
+      description: 'If `true`, the badge is checked otherwise it is unchecked.',
+      if: {
+        arg: 'interactionType',
+        eq: 'checkable'
+      },
+      control: {
+        type: 'boolean',
+      },
+      table: {
+        category: 'Interactions',
+      },
+    },
+    dismissed: {
+      name: 'Dismissed',
+      description: 'If `true`, the badge is removed from the page otherwise it is displayed.',
+      if: {
+        arg: 'interactionType',
+        eq: 'dismissible',
+      },
+      control: {
+        type: 'boolean',
+      },
+      table: {
+        category: 'Interactions',
+      },
+    },
+  },
+};
+
+export default meta;
+
+// DECORATORS
+function externalControl(story: any, { args }: StoryContext) {
+  const [_, updateArgs] = useArgs();
+
+  const button = html`
+    <a
+      href="#"
+      @click=${(e: Event) => {
+        e.preventDefault();
+        updateArgs({ dismissed: false });
+      }}
+    >Show badge</a>
+  `;
+
+  return html`
+    ${args.dismissed ? button : nothing}
+    ${story()}
+  `;
+}
+
+// RENDERER
+function getDefaultContent(args: Args) {
+  if (!args.nestedBadge) return html`${args.text}`;
+
+  return html`
+    <span>${args.text}</span>
```

Comment:
As suggested by @oliverschuerch:

```suggestion
    <span>${html`${args.text}`}</span>
```